src/cmd/services/m3query/config/config.go

```diff
@@ -746,6 +753,24 @@ type RPCConfiguration struct {
 	ReflectionEnabled bool `yaml:"reflectionEnabled"`
 }
 
+// PrometheusRemoteBackendConfiguration configures prometheus remote write backend.
+type PrometheusRemoteBackendConfiguration struct {
+	Endpoints       []PrometheusRemoteBackendEndpointConfiguration `yaml:"endpoints"`
+	RequestTimeout  *time.Duration                                 `yaml:"requestTimeout"`
+	ConnectTimeout  *time.Duration                                 `yaml:"connectTimeout"`
+	KeepAlive       *time.Duration                                 `yaml:"keepAlive"`
+	IdleConnTimeout *time.Duration                                 `yaml:"idleConnTimeout"`
+	MaxIdleConns    *int                                           `yaml:"maxIdleConns"`
+}
+
+// PrometheusRemoteBackendEndpointConfiguration configures single endpoint.
+type PrometheusRemoteBackendEndpointConfiguration struct {
+	Name       string        `yaml:"name"`
+	Address    string        `yaml:"address"`
+	Retention  time.Duration `yaml:"retention"`
+	Resolution time.Duration `yaml:"resolution"`
```

Comment:
I'm still wondering if I should make these required or optional. 0 kinda means raw unaggregated namespace. But should it be explicit in configuration?